### PR TITLE
A few minor changes

### PR DESCRIPTION
**Pull request by mlorbetske**
Shows the set of things that can be uninstalled when passing no arguments to the uninstall switch

Add a debug:version switch to show the version and commit hash

Prevent showing usage help when Initialize indicates that we should stop

### CODE REVIEW

File: src/Microsoft.TemplateEngine.Cli/CommandParsing/NewCommandInputCli.cs [hunk 1]
```diff
@@ -121,7 +122,7 @@ private void ParseArgs()
             _parseResult = parser.Parse(argsWithCommand.ToArray());
             _templateParamCanonicalToVariantMap = null;
 
-            IList<string> templateNameList = _parseResult.GetArgumentListAtPath(new[] { _commandName }).ToList();
+            IReadOnlyList<string> templateNameList = _parseResult.GetArgumentListAtPath(new[] { _commandName })?.ToList() ?? Empty<string>.List.Value;
```
Comment (seancpeters) [inline src/Microsoft.TemplateEngine.Cli/CommandParsing/NewCommandInputCli.cs]: GetArgumentListAtPath() can never return null, it'll return an empty list if there was nothing specified.

File: src/Microsoft.TemplateEngine.Cli/CommandParsing/NewCommandInputCli.cs [hunk 2]
```diff
@@ -213,11 +214,11 @@ public IReadOnlyList<string> Tokens
 
         public string BaselineName => _parseResult.GetArgumentValueAtPath(new[] { _commandName, "baseline" });
 
-        public IList<string> ExtraArgsFileNames => _parseResult.GetArgumentListAtPath(new[] { _commandName, "extra-args" }).ToList();
+        public IList<string> ExtraArgsFileNames => _parseResult.GetArgumentListAtPath(new[] { _commandName, "extra-args" })?.ToList();
```
Comment (seancpeters) [inline src/Microsoft.TemplateEngine.Cli/CommandParsing/NewCommandInputCli.cs]: Same comment as above - this'll never return null.

File: src/Microsoft.TemplateEngine.Edge/Settings/InstallUnitDescriptorCache.cs [hunk 3]
```diff
@@ -44,6 +50,39 @@ public bool TryAddDescriptorForLocation(Guid mountPointId)
                     return false;
                 }
 
+                string uninstallString = mountPoint.Info.Place;
+
+                //Adjust the uninstall string for NuGet packages if needed
+                if (uninstallString.EndsWith(".nupkg", StringComparison.OrdinalIgnoreCase))
```
Comment (seancpeters) [inline src/Microsoft.TemplateEngine.Edge/Settings/InstallUnitDescriptorCache.cs]: This logic should be in the NupkgInstallUnitDescriptorFactory.
There should be no descriptor implementation specific concerns in this file

File: src/Microsoft.TemplateEngine.Edge/Settings/InstallUnitDescriptorCache.cs [hunk 4]
```diff
@@ -44,6 +50,39 @@ public bool TryAddDescriptorForLocation(Guid mountPointId)
...
+                        }
+                        finally
+                        {
+                            _environmentSettings.SettingsLoader.ReleaseMountPoint(parentMountPoint);
+                        }
+                    }
+
+                    if (isFile)
+                    {
+                        if (NupkgInstallUnitDescriptorFactory.TryGetPackageInfoFromNuspec(mountPoint, out string packageName, out string _))
```
Comment (seancpeters) [inline src/Microsoft.TemplateEngine.Edge/Settings/InstallUnitDescriptorCache.cs]: "out string _" - even if this is throwaway, it should get a meaningful name.

**Comment by seancpeters** [comment]
@dotnet-bot test CentOS7.1 x64 Debug Build please
@dotnet-bot test Linux x64 Release Build please
@dotnet-bot test Ubuntu16.04 x64 Debug Build please

File: src/Microsoft.TemplateEngine.Edge/Settings/InstallUnitDescriptorCache.cs (hunk 3 above)
Comment (mlorbetske) [reply src/Microsoft.TemplateEngine.Edge/Settings/InstallUnitDescriptorCache.cs]: I'd prefer to not make this change now as the right way of doing it requires a bunch of additional work in factoring Installer. We can file another issue if needed to clean it up

**Comment by seancpeters** [comment]
@dotnet-bot test Linux x64 Release Build please
@dotnet-bot test Ubuntu16.04 x64 Debug Build please

File: src/Microsoft.TemplateEngine.Edge/Settings/InstallUnitDescriptorCache.cs (hunk 3 above)
Comment (seancpeters) [reply src/Microsoft.TemplateEngine.Edge/Settings/InstallUnitDescriptorCache.cs]: agreed... deferring

**Comment by seancpeters** [comment]
@dotnet-bot test Linux x64 Release Build please

**Comment by mlorbetske** [comment]
@dotnet-bot test Windows_NT x64 Release Build please

**Pull request merged**